docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -25,63 +30,132 @@ This setup causes problems when:
+
+Suggestions trust domain on global: `spiffe://<mesh>.<KumaClusterId>.kuma.io/` - is generated on Global, to ensure uniquness
 
 ### As a Mesh operator, I want to be able to migrate some workloads to another trust domain without interrupting traffic.
 
 I have a workloads in a trust domain `example.com` and I want to move them to `example.org`. I should be able to do it and don't break the traffic.
 
+### As a user, I want certificate to be valid for a short period of time
+
+Based on the [specification](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-workload-api) certificates should be short-lived.
+
+## Future User Stories (things that are going to be implemented in future releases)
+
+### As a user, I want to have a SPIFFE-compliant SAN in a non-Kubernetes environment
+
+It would be beneficial to support connectivity between different environments. Unfortunately, our services currently lack proper identity on Universal, which makes this challenging.
+
+### As a user, I want to federate different trust domains using the Federation API exposed by the control plane
+
+It should be possible to federate different trust domains without creating additional resources, only by using the federation endpoint exposed by the control plane. This would be a valuable feature, enabling interoperability between different SPIFFE-compliant systems.
+
+### As a user, I want to use an intermediate CA instead of a root CA on the zone
+
+When a user provides a root CA on the global control plane, we could generate a separate intermediate CA for each connecting zone. This would offer a stronger security model, ensuring that if a certificate is compromised in one zone, it does not impact other zones.
+
+### As a user, I want to Trust from my zone to be synced to other zones automatically.
+
+Trust could be automatically generated by the control plane and propagated to other zones. While this would improve automation and reduce configuration overhead, it is not critical at this stage and can be addressed in a future iteration.
+
+### As a user, I want to use all available CA providers from legacy mTLS in new mTLS setup
+
+Kuma currently supports various CA (Certificate Authority) backends for identity management. The new mTLS system should at least support the same set of CA providers, to ensure a smooth transition and preserve compatibility with existing deployments.
+
+### As a user, I want to migrate from Kuma non-SPIFFE complient certificates to SPIRE
+
+We should allow users to migrate from Kuma issued certificates, which are not SPIFFE-compliant, to using SPIRE. This may require an intermediate step to first migrate to SPIFFE-compliant certificates, but the details will be specified later.
+
+### As a user, I want to automatically rotate the CA
+
+This could be straightforward to implement, but we shouldn't prioritize it at the moment.
```

Comment:
I think it makes sense to add @lahabana comment here https://github.com/kumahq/kuma/pull/13699/files/7bf39d3fd8dd74da03cfbdf22f1c832d120f9fff..03927ffdadd7e6e30b897d0b7d6a8ef431502078#r2152062929

Reply:
It seems like SPIRE description, I would stay without specific description since it's not implementation but Context and Problem statement